x-pack/plugins/lens/public/xy_visualization/visualization.tsx

```diff
@@ -560,11 +567,28 @@ function checkXAccessorCompatibility(
       }),
     });
   }
+  if (state.layers.length > 1 && (hasDateHistogramSet || hasNumberHistogram) && hasOrdinalAxis) {
+    errors.push({
+      shortMessage: i18n.translate('xpack.lens.xyVisualization.dataTypeFailureXShort', {
+        defaultMessage: `Wrong data type for {axis}.`,
+        values: {
+          axis: getAxisName('x', { isHorizontal: isHorizontalChart(state.layers) }),
+        },
+      }),
+      longMessage: i18n.translate('xpack.lens.xyVisualization.dataTypeFailureXOrdinalLong', {
+        defaultMessage: `Data type mismatch for the {axis}. Cannot mix interval and ordinal scale types.`,
```

Comment:
The terms `interval` and `ordinal` are a little confusing, maybe it's enough to say `Data type mismatch for the {axis}, use a different function`?